service/src/main/java/uk/gov/mca/beacons/api/registration/application/RegistrationReadOnlyService.java

```diff
@@ -79,7 +81,66 @@ public Registration getByBeaconId(BeaconId beaconId) {
       .build();
   }
 
-  public List<Note> getNonSyetemNotesByBeaconId(BeaconId beaconId) {
+  public Registration getRegistrationFromBeaconBackupItem(
+    BeaconBackupItem beaconBackupItem
+  ) {
+    Registration registration = Registration.builder().build();
+    registration =
+      hydrateBeaconFromBeaconBackupItem(registration, beaconBackupItem);
+    BeaconId beaconId = registration.getBeacon().getId();
+
+    BeaconOwner beaconOwner = beaconOwnerRepository
+      .findBeaconOwnerByBeaconId(beaconId)
+      .orElse(null);
+    List<BeaconUse> beaconUses = beaconUseRepository.findBeaconUsesByBeaconId(
+      beaconId
+    );
+    List<EmergencyContact> emergencyContacts = emergencyContactRepository.findEmergencyContactsByBeaconId(
+      beaconId
+    );
+
+    registration.setBeaconOwner(beaconOwner);
```

Comment:
Would it make sense to have this as a builder method instead too?

Fetching the additional stuff above, then 

return Registration.builder()
.beacon(buildBeaconFromBackupItem(backupItem))
.beaconOwner(beaconOwner)
.beaconUses(beaconUses)
.emergencyContacts(emergencyContacts)
.build();

Reply:
Maybe pulling Beacon beacon = buildBeaconFromBackupItem(backupItem) above or something for debugging purposes

Reply:
See above re builder